src/content/en/updates/2017/12/chrome-loadtimes-deprecated.md

```diff
@@ -0,0 +1,366 @@
+  }
+}
+```
+
+### `startLoadTime`
+
+```
+function startLoadTime() {
+  // If the browser supports the Navigation Timing 2 and HR Time APIs, use
+  // them, otherwise fall back to the Navigation Timing 1 API.
+  if (window.PerformanceNavigationTiming && performance.timeOrigin) {
+    const ntEntry = performance.getEntriesByType('navigation')[0];
+    return (ntEntry.startTime + performance.timeOrigin) / 1000;
+  } else {
+    return performance.timing.navigationStart / 1000;
+  }
+}
+```
+
+### `commitLoadTime`
+
+```
+function commitLoadTime() {
+  // If the browser supports the Navigation Timing 2 and HR Time APIs, use
+  // them, otherwise fall back to the Navigation Timing 1 API.
+  if (window.PerformanceNavigationTiming && performance.timeOrigin) {
+    const ntEntry = performance.getEntriesByType('navigation')[0];
+    return (ntEntry.responseStart + performance.timeOrigin) / 1000;
+  } else {
+    return performance.timing.responseStart / 1000;
+  }
+}
+```
+
+### `finishDocumentLoadTime`
+
+```
+function finishDocumentLoadTime() {
+  // If the browser supports the Navigation Timing 2 and HR Time APIs, use
+  // them, otherwise fall back to the Navigation Timing 1 API.
+  if (window.PerformanceNavigationTiming && performance.timeOrigin) {
+    const ntEntry = performance.getEntriesByType('navigation')[0];
+    return (ntEntry.domContentLoadedEventEnd + performance.timeOrigin) / 1000;
+  } else {
+    return performance.timing.domContentLoadedEventEnd / 1000;
+  }
+}
+```
+
+### `finishLoadTime`
+
+```
+function finishLoadTime() {
+  // If the browser supports the Navigation Timing 2 and HR Time APIs, use
+  // them, otherwise fall back to the Navigation Timing 1 API.
+  if (window.PerformanceNavigationTiming && performance.timeOrigin) {
+    const ntEntry = performance.getEntriesByType('navigation')[0];
+    return (ntEntry.loadEventEnd + performance.timeOrigin) / 1000;
+  } else {
+    return performance.timing.loadEventEnd / 1000;
+  }
+}
+```
+
+### `firstPaintTime`
+
+```
+function firstPaintTime() {
+  if (window.PerformancePaintTiming) {
+    const fpEntry = performance.getEntriesByType('paint')[0];
+    return (fpEntry.startTime + performance.timeOrigin) / 1000;
+  }
+}
+```
+
+### `firstPaintAfterLoadTime`
+
+```
+function firstPaintTimeAfterLoad() {
+  // This was never actually implemented and always returns 0.
+  return 0;
+}
+```
+
+### `navigationType`
+
+```
+function navigationType() {
+  if (window.PerformanceNavigationTiming) {
+    const ntEntry = performance.getEntriesByType('navigation')[0];
+    return ntEntry.type;
+  }
+}
+```
+
+### `wasFetchedViaSpdy`
+
+```
+function wasFetchedViaSpdy() {
+  // SPDY is deprecated in favor of HTTP/2, but this implementation returns
+  // true for HTTP/2 or HTTP2+QUIC/39 as well.
+  if (window.PerformanceNavigationTiming) {
+    const ntEntry = performance.getEntriesByType('navigation')[0];
+    return ['h2', 'hq'].includes(ntEntry.nextHopProtocol);
```

Comment:
Not 100% sure that we should be including `hq` here.. But I can see arguments both ways. I guess this is OK.

Reply:
Yeah, I wasn't sure either, but this does make the function return what is actually returned by `chrome.loadTimes().wasFetchedViaSpdy`, so I kept it for consistency.